net/xia/ppal_ether/main.c

```diff
@@ -0,0 +1,882 @@
+#include <linux/module.h>
+#include <net/xia_dag.h>
+#include <net/xia_output.h>
+#include <net/xia_vxidty.h>
+#include <linux/rwlock.h>
+#include <net/xia_ether.h>
+
+/* ETHER_FIB table internal operations */
+const struct xia_ppal_rt_iops *ether_rt_iops = &xia_ppal_list_rt_iops;
+
+/* ETHER local table operations */
+static int local_newroute(struct xip_ppal_ctx *ctx,
+			  struct fib_xid_table *xtbl,
+			  struct xia_fib_config *cfg)
+{
+	struct fib_xid_ether_local *leid;
+	int rc;
+
+	/* Allocate memory to new local ether xid*/
+	leid = ether_rt_iops->fxid_ppal_alloc(sizeof(*leid), GFP_KERNEL);
+	// If not enough space
+	if (!leid)
+		return -ENOMEM;
+	// Initialize the fib_xid inside the local ether xid
```

Comment:
Please remove the comment, since one can know what it does by looking at the definition of ``fxid_init()``, and it doesn't add more information.